src/Arith.jl

```diff
@@ -149,39 +146,71 @@ Base.iszero(x::CycloSum) = iszero(x.summands[1])  # isone(length(x.summands)) is
 struct CycloFrac{T} <: Cyclotomic{T}
 	numerator::CycloSum{T}
 	denominator::CycloSum{T}
-	function CycloFrac(numerator::CycloSum{T}, denominator::CycloSum{T}; simplify::Bool=true) where T<:NfPoly
+	exceptions::Set{ParameterException{T}}
+	function CycloFrac(numerator::CycloSum{T}, denominator::CycloSum{T}, exceptions::Set{ParameterException{T}}; simplify::Bool=true) where T<:NfPoly
 		if simplify
 			if iszero(numerator)
-				return new{T}(numerator, one(denominator))
+				return new{T}(numerator, one(denominator), exceptions)
 			elseif numerator == denominator
 				o=one(denominator)
-				return new{T}(o, o)
+				return new{T}(o, o, exceptions)
 			else
-				return new{T}(numerator, denominator)
+				return new{T}(numerator, denominator, exceptions)
 			end
 		else
-			return new{T}(numerator, denominator)
+			return new{T}(numerator, denominator, exceptions)
 		end
 	end
 end
+function add_exception!(a::CycloFrac{T}, exception::ParameterException{T}) where T<:PolyRingElem
+	push!(a.exceptions, exception)
+end
+"""
+    shrink(a::CycloFrac{<:NfPoly})
+
+Remove exceptions from `a` that follow from the others. And try to simplify the representation of `a`.
+"""
+function shrink(a::CycloFrac{T}) where T <: NfPoly  # TODO Move this to the constructor of CycloFrac?
+	new_numerator=a.numerator
+	new_denominator=a.denominator
+	if isone(length(a.numerator.summands)) && isone(length(a.denominator.summands))
+		principal_numerator=a.numerator.summands[1]
+		principal_denominator=a.denominator.summands[1]
+		if iszero(principal_numerator.argument) && iszero(principal_denominator.argument)
+			b=principal_numerator.modulus//principal_denominator.modulus
+			o=one(a.numerator)
+			new_numerator=o*numerator(b)
+			new_denominator=o*denominator(b)
+		end
+	end
+	return CycloFrac(new_numerator, new_denominator, shrink(a.exceptions), simplify=false)
+end
 function Base.show(io::IO, z::CycloFrac)
+	if isempty(z.exceptions)
+		exceptions=""
+	else
+		exceptions="\nWith exceptions:"
+		for exception in z.exceptions
+			exceptions*="\n  $(exception)"
+		end
+	end
 	if isone(length(z.numerator.summands))
 		if isone(length(z.denominator.summands))
-			print(io, "$(z.numerator)//$(z.denominator)")
+			print(io, "$(z.numerator)//$(z.denominator)$(exceptions)")
 		else
-			print(io, "$(z.numerator)//($(z.denominator))")
+			print(io, "$(z.numerator)//($(z.denominator))$(exceptions)")
 		end
 	else
 		if isone(length(z.denominator.summands))
-			print(io, "($(z.numerator))//$(z.denominator)")
+			print(io, "($(z.numerator))//$(z.denominator)$(exceptions)")
 		else
-			print(io, "($(z.numerator))//($(z.denominator))")
+			print(io, "($(z.numerator))//($(z.denominator))$(exceptions)")
 		end
```

Comment:
simpler patch: just do `println(io, exceptions)` at the end. That will also be more amenable to future changes to e.g. adopt the AbstractAlgebra/OSCAR printing modes (terse/oneline/supercompact, see [here](https://docs.oscar-system.org/dev/DeveloperDocumentation/printing_details/))

Reply:
Shouldn't it be just `print` not `println`?

Reply:
`print` seems to do the trick.